### PR TITLE
feat. add lepton ai

### PR DESCRIPTION
**Pull request by Ajay-Satish-01**
Code Quality new feature 

## Author Description

**Title:** 
- Add Lepton AI support

**Related Issues:** 
- #671

**Title:** 
- Add Lepton AI support

### 🔄 What Changed
- Added Lepton AI as a new provider in the gateway
- Implemented API configuration for Lepton endpoints
- Added support for chat completion, text completion, and transcription services
- Implemented stream handling for both chat and text completions

### 🔍 Impact of the Change
- Expands the gateway's capabilities by integrating Lepton AI as a new provider
- Allows users to access Lepton's AI services through the existing gateway interface
- Follows established patterns for provider integration, ensuring consistency

### 📁 Total Files Changed
- 7 files changed (495 additions, 0 deletions)
- 2 files modified: globals.ts, providers/index.ts
- 5 files added: providers/lepton/* (api.ts, chatComplete.ts, complete.ts, createTranscription.ts, index.ts)

### 🧪 Test Added
- N/A

### 🔒 Security Vulnerabilities
- No security vulnerabilities identified
- Proper API key handling through Authorization header

**Related Issues:** 
- #671

## Quality Recommendations

1. Add unit tests for the Lepton provider integration to ensure proper functionality

2. Add error handling for edge cases in stream chunk parsing

3. Add JSDoc comments to explain the purpose and parameters of key functions

4. Consider adding validation for Lepton-specific parameters

5. Add logging for API request failures to aid in debugging

### CODE REVIEW

File: src/providers/lepton/createTranscription.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
+import { LEPTON } from '../../globals';
+import { ErrorResponse, ProviderConfig } from '../types';
+import { OpenAIErrorResponseTransform } from '../openai/utils';
+
+export const LeptonCreateTranscriptionConfig: ProviderConfig = {
```
Comment (narengogi) [inline src/providers/lepton/createTranscription.ts]: Hey @Ajay-Satish-01 does the request body of the `/api/v1/audio/transcriptions` route take a `json` payload or a `multipart/formdata`?

configs are supported only for JSON payloads transformation, and this one is unused

in this case, you can simply remove this config and in `lepton/index.ts` you can specify an empty object for `createTranscriptions` request transform, like this:
```js
const LeptonConfig: ProviderConfigs = {
  chatComplete: LeptonChatCompleteConfig,
  complete: LeptonCompleteConfig,
  createTranscription: {},
  api: LeptonAPIConfig,
  responseTransforms: {
    chatComplete: LeptonChatCompleteResponseTransform,
    'stream-chatComplete': LeptonChatCompleteStreamChunkTransform,
    complete: LeptonCompleteResponseTransform,
    'stream-complete': LeptonCompleteStreamChunkTransform,
    createTranscription: LeptonCreateTranscriptionResponseTransform,
  },
};
```

**Review by narengogi — CHANGES_REQUESTED** [review]
LGTM, only one minor comment @Ajay-Satish-01

File: src/providers/lepton/createTranscription.ts (hunk 1 above)
Comment (Ajay-Satish-01) [reply src/providers/lepton/createTranscription.ts]: @narengogi Implemented the suggested approach.

**Review by narengogi — APPROVED** [review]
